### PR TITLE
Feature/adding location

### PR DESCRIPTION
**Pull request by delcroip**

### CODE REVIEW

File: individual/tests/graphql_mutation_group_test.py [hunk 1]
```diff
@@ -722,17 +732,17 @@ def test_remove_individuals_from_group_row_security(self):
...
-        response = self.query(
-            query_str.replace(
-                str(group_individual_a.id),
-                str(group_individual_b.id)
-            ), headers={"HTTP_AUTHORIZATION": f"Bearer {self.dist_b_user_token}"}
-        )
-        content = json.loads(response.content)
-        id = content['data']['removeIndividualFromGroup']['internalId']
-        self.assert_mutation_success(id)
+        # group_no_loc = create_group(self.admin_user.username)
```
Comment (delcroip) [inline individual/tests/graphql_mutation_group_test.py]: @weilu this test create a wierd error Group is not callable, please have a look I have sick of this module after fixing test for several days

Comment (weilu) [reply individual/tests/graphql_mutation_group_test.py]: I uncommented the test case and it passed for me locally with the location module pointed to the `feature/is-allowed` branch. Can you paste the full stacktrace?

File: individual/tests/graphql_mutation_group_test.py [hunk 2]
```diff
@@ -613,11 +624,7 @@ def test_edit_individual_in_group_row_security(self):
             headers={"HTTP_AUTHORIZATION": f"Bearer {self.admin_token}"}
         )
         content = json.loads(response.content)
-        id = content['data']['editIndividualInGroup']['internalId']
-        self.assert_mutation_error(id, 'mutation.individual_group_village_mismatch')
```
Comment (weilu) [inline individual/tests/graphql_mutation_group_test.py]: @delcroip This test case shouldn't have been deleted as it catches the case that an imis admin user can manipulate any individual's group assignment but we should validate to ensure the individual and their assigned group's locations are consistent.

File: individual/tests/graphql_mutation_group_test.py [hunk 3]
```diff
@@ -175,24 +180,27 @@ def test_update_group_row_security(self):
 
         # SP officer B cannot update group for district A
         response = self.query(query_str)
+        content = json.loads(response.content)
+        internal_id = content['data']['updateGroup']['internalId']
+        self.assert_mutation_error(internal_id, 'mutation.authentication_required')
```
Comment (weilu) [inline individual/tests/graphql_mutation_group_test.py]: This shouldn't be necessary as the anonymous user case is already tested in the test above. I see why you added them – probably due to the extraneous line I have above where the response was not checked. I'll push a change to remove this and similar changes below.

**Review by weilu — CHANGES_REQUESTED** [review]
@delcroip The change to use `LocationManager.is_allowed` instead of `Location.get_queryset` introduces the following issues: 
1. it no longer considers and respects `settings.ROW_SECURITY`,
2.  users with `user.is_imis_admin = True` now is not permitted to mutate any individual or group with location. Case 2 is actually covered by a test case you removed (see my comment in code).

In other words, `Location.get_queryset` offers more functionalities needed than that `LocationManager.is_allowed` is currently offering. In the spirit of DRY, I'd suggest not replicating the checks inside `is_allowed`, but considering pushing the caching logic into `LocationManager.allowed` which is used by `Location.get_queryset` and keep using `Location.get_queryset` for location authorization checking.

File: individual/tests/graphql_mutation_group_test.py (hunk 3 above)
Comment (delcroip) [reply individual/tests/graphql_mutation_group_test.py]: I added it to force the execution of the mutation, if not error could happen later in the process, if we remove the get_mutation_result we should also remove the query

Comment (weilu) [reply individual/tests/graphql_mutation_group_test.py]: Yes, I removed the query in https://github.com/openimis/openimis-be-individual_py/pull/129/commits/324c578daf909a9b374d37d20ed1d8b04907a4db

File: individual/tests/graphql_mutation_group_test.py [hunk 4]
```diff
@@ -474,7 +474,7 @@ def test_add_individual_to_group_row_security(self):
         )
         content = json.loads(response.content)
         internal_id = content['data']['addIndividualToGroup']['internalId']
-        self.assert_mutation_error(internal_id, _('unauthorized.location'))
+        self.assert_mutation_success(internal_id)
```
Comment (weilu) [inline individual/tests/graphql_mutation_group_test.py]: @delcroip why has this been changed to success? Shouldn't we prevent such a scenario to keep location data consistent between individual and their group?

**Review by weilu — CHANGES_REQUESTED** [review]

**Review by weilu — APPROVED** [review]

**Pull request merged**